tools/c7n_azure/c7n_azure/resources/sqldatabase.py

```diff
@@ -30,3 +40,196 @@ class resource_type(ArmResourceManager.resource_type):
             manager_name='sqlserver',
             annotate_parent=True
         )
+
+
+class BackupRetentionPolicyFilter(Filter):
+
+    schema = type_schema(
+        'backup-retention-policy',
+        **{
+            'op': {'enum': list(scalar_ops.keys())}
+        }
+    )
+
+    def __init__(self, operations_property, retention_limit, data, manager=None):
+        super(BackupRetentionPolicyFilter, self).__init__(data, manager)
+        self.operations_property = operations_property
+        self.retention_limit = retention_limit
+
+    def get_retention_from_policy(self, retention_policy):
+        raise NotImplementedError()
+
+    def process(self, resources, event=None):
+        resources, exceptions = ThreadHelper.execute_in_parallel(
+            resources=resources,
+            event=event,
+            execution_method=self._process_resource_set,
+            executor_factory=self.executor_factory,
+            log=log
+        )
+        if exceptions:
+            raise exceptions[0]
+        return resources
+
+    def _process_resource_set(self, resources, event):
+        client = self.manager.get_client()
+        get_operation = getattr(client, self.operations_property).get
+        matched_resources = []
+
+        for resource in resources:
+            match = self._process_resource(resource, get_operation)
+            if match:
+                matched_resources.append(resource)
+        return matched_resources
+
+    def _process_resource(self, i, get_operation):
+        retention_policy = self._get_backup_retention_policy(i, get_operation)
```

Comment:
let's cache returned data. You can use `c7n` prefix & some way to specify if it is short\long term and its type (monthly, daily etc)

Reply:
I have already run into a need to reference this cache in a separate branch I'm working on :) 

I'll work that implementation into this.